arch/xtensa/soc/esp32/esp32-mp.c

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <kernel_structs.h>
+#include <soc.h>
+
+#define _REG(base, off) (*(volatile u32_t *)((base) + (off)))
+
+#define RTC_CNTL_BASE             0x3ff48000
```

Comment:
Please use the register macros from esp-idf. All ESP32 drivers are using them; I'd like to keep the consistency.

Reply:
Sorry, my reply must have been in the other PR.  The problem is that the existing API for these registers is sort of a mess, it doesn't express the register as an lvalue and wants to use a whole bunch of setter/getter macroness, and IIRC I had trouble making that work.  But I'll investigate again.